shap/plots/_group_difference.py

```diff
@@ -52,36 +52,34 @@ def group_difference(shap_values, group_mask, feature_names=None, xlabel=None, x
 
     if max_display is not None:
         inds = inds[:max_display]
-    # draw the figure
-    figsize = [6.4, 0.2 + 0.9 * len(inds)]
-    pl.figure(figsize=figsize)
+    if ax:
+        # Disable plotting out if an ax has been provided
+        show = False
```

Comment:
I _think_ I see your reasoning for this, but for the sake of record, do you mind explaining why we choose to explicitly set `show` to False and override the user's argument?

Reply:
In short, my thinking is that if `ax` is provided, then the corresponding `fig` has already been instantiated outside of this function and showing the plot should be handled there after all axes of the figure have been filled in with plots.

Reply:
I see where you're coming from. I'm just not sure if overriding the user's input is the best user experience (e.g. if the user explicitly states `show=True, ax=ax`, and the figure isn't shown, the user might be surprised as to why that is.

It's alright for now, we can standardize this behaviour across all plotting functions in a future PR.